@mattclay here are some of my findings

`fetch_url` calls `open_url` and internally calls `urllib_request.urlopen`

On Python 2.6 

```
Python 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-17)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2 as urllib_request
>>> url = "https://api.ipify.org?format=json"
>>> urllib_request.urlopen(url).read()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.6/urllib2.py", line 126, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib64/python2.6/urllib2.py", line 397, in open
    response = meth(req, response)
  File "/usr/lib64/python2.6/urllib2.py", line 510, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python2.6/urllib2.py", line 435, in error
    return self._call_chain(*args)
  File "/usr/lib64/python2.6/urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.6/urllib2.py", line 518, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 400: Bad Request
```

Whereas, it works perfect on Python 2.7 

```
Python 2.7.10 (default, Jul 30 2016, 18:31:42)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2 as urllib_request
>>> url = "https://api.ipify.org?format=json"
>>> urllib_request.urlopen(url).read()
'{"ip":"*.*.10.110"}'
```

This leaves us with options

1. Stop fixing for Python 2.6 as it is not supported anymore, there by skipping CentOS6 test,
2. Replace `fetch_url` with `requests` in `ipify_facts` module,

Let me know your preference.

--- a/lib/ansible/modules/net_tools/ipify_facts.py
+++ b/lib/ansible/modules/net_tools/ipify_facts.py
@@ -44,6 +44,12 @@ options:
     required: false
     default: 10
     version_added: "2.3"
+  validate_certs:
+    description:
+      - When set to C(NO), SSL certificates will not be validated.
+    required: false
+    default: "yes"
+    version_added: "2.4"
 notes:
   - "Visit https://www.ipify.org to get more information."
 '''
@@ -80,6 +86,7 @@ except ImportError:
 
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_text
 
 
 class IpifyFacts(object):
@@ -92,21 +99,23 @@ class IpifyFacts(object):
         result = {
             'ipify_public_ip': None
         }
-        (response, info) = fetch_url(module=module, url=self.api_url + "?format=json" , force=True, timeout=self.timeout)
+        (response, info) = fetch_url(module=module, url=self.api_url + "?format=json", force=True, timeout=self.timeout)
 
         if not response:
             module.fail_json(msg="No valid or no response from url %s within %s seconds (timeout)" % (self.api_url, self.timeout))
 
-        data = json.loads(response.read())
+        data = json.loads(to_text(response.read()))
         result['ipify_public_ip'] = data.get('ip')
         return result
+
 
 def main():
     global module
     module = AnsibleModule(
-        argument_spec = dict(
-            api_url=dict(default='https://api.ipify.org'),
+        argument_spec=dict(
+            api_url=dict(default='https://api.ipify.org/'),
             timeout=dict(type='int', default=10),
+            validate_certs=dict(type='bool', default=True),
         ),
         supports_check_mode=True,
     )

--- a/test/integration/targets/ipify_facts/aliases
+++ b/test/integration/targets/ipify_facts/aliases
@@ -1,0 +1,1 @@
+posix/ci/group2

--- a/test/integration/targets/ipify_facts/tasks/main.yml
+++ b/test/integration/targets/ipify_facts/tasks/main.yml
@@ -1,0 +1,39 @@
+# Test code for the ipify_facts
+# (c) 2017, Abhijeet Kasurde <akasurde@redhat.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+- debug: var=ansible_distribution
+- debug: var=ansible_distribution_version
+
+- set_fact:
+    validate_certs: false
+  when: (ansible_distribution == "MacOSX" and ansible_distribution_version == "10.11.1")
+
+- name: get information about current IP using ipify facts
+  ipify_facts:
+    timeout: 30
+    validate_certs: "{{ validate_certs }}"
+  register: external_ip
+- debug: var="{{ external_ip }}"
+
+- name: check if task was successful
+  assert:
+    that:
+      - "{{ external_ip.changed == false }}"
+      - "{{ external_ip['ansible_facts'] is defined }}"
+      - "{{ external_ip['ansible_facts']['ipify_public_ip'] is defined }}"

--- a/test/integration/targets/ipify_facts/vars/main.yml
+++ b/test/integration/targets/ipify_facts/vars/main.yml
@@ -1,0 +1,2 @@
+---
+validate_certs: true

--- a/test/sanity/pep8/legacy-files.txt
+++ b/test/sanity/pep8/legacy-files.txt
@@ -323,7 +323,6 @@ lib/ansible/modules/network/iosxr/iosxr_command.py
 lib/ansible/modules/network/iosxr/iosxr_config.py
 lib/ansible/modules/network/iosxr/iosxr_facts.py
 lib/ansible/modules/network/iosxr/iosxr_system.py
-lib/ansible/modules/net_tools/ipify_facts.py
 lib/ansible/modules/net_tools/ipinfoio_facts.py
 lib/ansible/modules/network/lenovo/cnos_conditional_template.py
 lib/ansible/modules/network/lenovo/cnos_template.py

